### PR TITLE
OCP-24488: Fix for RHEL7 workers

### PR DESCRIPTION
**Pull request by rbbratta**
Now that I know how to create RHEL workers I can test on RHEL.

It turns out that the RHEL7 `bridge` JSON VLAN is output is different
from RHCOS

RHCOS JSON:
```json
[{"ifname":"bridge3","vlans":[{"vlan":1,"flags":["PVID","Egress Untagged"]}]},{"ifname":"veth66451995","vlans":[{"vlan":1,"flags":["PVID","Egress Untagged"]}]}]
```
RHEL7 JSON
```json
{"bridge3":[{"vlan":1,"flags":["PVID","EgressUntagged"]}],"vethb26eb609":[{"vlan":1,"flags":["PVID","EgressUntagged"]}]}
```

It looks like we can convert the loop to check `b[1]` first in case it is an Array.  Otherwise try `b["vlans"]`

### CODE REVIEW

**Comment by rbbratta** [comment]
```ruby
    Given CNI vlan info is obtained on the "<%= cb.pod.node_name %>" node                                                       # features/step_definitions/networking.rb:698
      [00:33:11] INFO> Shell Commands: oc debug  --kubeconfig=.kubeconfig -n tests-rbrattai-rbrattai --image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:8f4882cff3c2f9521215eac681c5abda42876e3e955431c1387fb457940b8344 node/ip-10-0-48-178.us-east-2.compute.internal -- chroot /host/ bash -c cd\ \'/tmp/workdir/rbrattai-rbrattai\''
      '/sbin/bridge\ -j\ vlan\ show
      {
          "bridge3": [{
                  "vlan": 1,
                  "flags": ["PVID","Egress Untagged"
                  ]
              }
          ],
          "veth36a7e85b": [{
                  "vlan": 1,
                  "flags": ["PVID","Egress Untagged"
                  ]
              }
          ]
      }
      
      STDERR:
      Starting pod/ip-10-0-48-178us-east-2computeinternal-debug ...
      To use host binaries, run `chroot /host`
      
      Removing debug pod ...
      
      [00:33:13] INFO> Exit Status: 0
    Then the step should succeed                                                                                                # features/step_definitions/common.rb:4
    And evaluation of `@result[:parsed]` is stored in the :vlans_after clipboard                                                # features/step_definitions/common.rb:128
      # check "ntagged" substring due to iproute2 changes,
      # iproute2 before v4.19.0 should output "untagged"
      # https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/commit/bridge/vlan.c?h=v5.3.0&id=0f36267485e30099a4f735c3aadfa58b5efa1918
      # after v4.19.0 it should output "Egress Untagged"
    Given the number of bridge PVID 1 VLANs matching "ntagged" added between the :vlans_before and :vlans_after clipboards is 2 # features/step_definitions/networking.rb:705
      [00:33:14] INFO> added_bridges: #<Set: {["bridge3", [{"vlan"=>1, "flags"=>["PVID", "Egress Untagged"]}]], ["veth36a7e85b", [{"vlan"=>1, "flags"=>["PVID", "Egress Untagged"]}]]}>
      [00:33:14] INFO> === After Scenario: Create pod with Multus bridge CNI plugin without vlan ===
```

```ruby
    Given CNI vlan info is obtained on the "<%= cb.pod.node_name %>" node                                                       # features/step_definitions/networking.rb:698
      [00:16:57] INFO> Shell Commands: oc debug  --kubeconfig=kubeconfig -n tests-rbrattai-rbrattai --image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:8f4882cff3c2f9521215eac681c5abda42876e3e955431c1387fb457940b8344 node/ip-10-0-55-30.us-east-2.compute.internal -- chroot /host/ bash -c cd\ \'/tmp/workdir/rbrattai-rbrattai\''
      '/sbin/bridge\ -j\ vlan\ show
      [{"ifname":"bridge3","vlans":[{"vlan":1,"flags":["PVID","Egress Untagged"]}]},{"ifname":"veth2fd03362","vlans":[{"vlan":1,"flags":["PVID","Egress Untagged"]}]}]
      
      STDERR:
      Starting pod/ip-10-0-55-30us-east-2computeinternal-debug ...
      To use host binaries, run `chroot /host`
      
      Removing debug pod ...
      
      [00:16:59] INFO> Exit Status: 0
    Then the step should succeed                                                                                                # features/step_definitions/common.rb:4
    And evaluation of `@result[:parsed]` is stored in the :vlans_after clipboard                                                # features/step_definitions/common.rb:128
      # check "ntagged" substring due to iproute2 changes,
      # iproute2 before v4.19.0 should output "untagged"
      # https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/commit/bridge/vlan.c?h=v5.3.0&id=0f36267485e30099a4f735c3aadfa58b5efa1918
      # after v4.19.0 it should output "Egress Untagged"
    Given the number of bridge PVID 1 VLANs matching "ntagged" added between the :vlans_before and :vlans_after clipboards is 2 # features/step_definitions/networking.rb:705
      [00:16:59] INFO> added_bridges: #<Set: {{"ifname"=>"bridge3", "vlans"=>[{"vlan"=>1, "flags"=>["PVID", "Egress Untagged"]}]}, {"ifname"=>"veth2fd03362", "vlans"=>[{"vlan"=>1, "flags"=>["PVID", "Egress Untagged"]}]}}>
      [00:16:59] INFO> === After Scenario: Create pod with Multus bridge CNI plugin without vlan ===
```

**Comment by anuragthehatter** [comment]
@rbbratta Thanks. This logic looks good to me. Hopefully this stays same on Rhel 8 as well which will hit soon in our mix nodes testing. The last thing would be to check if t still works fine on complete RCOS cluster else its good to go. That the CRIO config brought us this pain else i was thinking to just check for 2 untagged presence against veth and bridge. The above logs on rhel node?

**Comment by rbbratta** [comment]
I tested on all the nodes of a 4.6 SDN ocp4-rhel-scaleup cluster.

RHEL7
```json
      '/sbin/bridge\ -j\ vlan\ show
      {
          "bridge3": [{
                  "vlan": 1,
                  "flags": ["PVID","Egress Untagged"
                  ]
              }
          ],
          "veth36a7e85b": [{
                  "vlan": 1,
                  "flags": ["PVID","Egress Untagged"
                  ]
              }
          ]
      }
````


RHCOS
```json
      [{"ifname":"bridge3","vlans":[{"vlan":1,"flags":["PVID","Egress Untagged"]}]},{"ifname":"veth2fd03362","vlans":[{"vlan":1,"flags":["PVID","Egress Untagged"]}]}]

```

**Comment by anuragthehatter** [comment]
Awesome. Thanks Ross.

/LGTM

File: features/step_definitions/networking.rb [hunk 1]
```diff
@@ -704,11 +704,19 @@
...
   logger.info("added_bridges: #{added_bridges}")
   mode = Regexp.new(mode)
   num_vlans = added_bridges.count { |b|
-    c = b["vlans"].count { |v|
+    # handle old RHEL7 bridge JSON and current RHCOS bridge JSON output
+    # RHCOS:
+    # [{"ifname":"bridge3","vlans":[{"vlan":1,"flags":["PVID","Egress Untagged"]}]},{"ifname":"veth66451995","vlans":[{"vlan":1,"flags":["PVID","Egress Untagged"]}]}]
+    # RHEL7
+    # {"bridge3":[{"vlan":1,"flags":["PVID","EgressUntagged"]}],"vethb26eb609":[{"vlan":1,"flags":["PVID","EgressUntagged"]}]}
+    # try b[1] first else cneck for the "vlans" key
```
Comment (weliang1) [inline features/step_definitions/networking.rb:717]: Typo, "cneck", should be "check"

Comment (rbbratta) [reply features/step_definitions/networking.rb:717]: https://github.com/openshift/verification-tests/pull/1629

**Pull request merged**